src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private RandomSize size = new RandomSize();
+
+    private Figure circle = new Circle(size.getSize(), colorSupplier.getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(size.getSize(),
+            size.getSize(), size.getSize(), colorSupplier.getRandomColor());
+    private Figure rectangle = new Rectangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+    private Figure square = new Circle(size.getSize(),

```

Comment:
It seems there is a mistake when creating the 'square' object. You're using the 'Circle' class instead of the 'Square' class.